File: smells/debugflag.js
const vscode = require('vscode');

var smell = {

    detect : (token) => {
                
        if(token.hasOwnProperty("line")) var lineno = token.line;
        if(token.hasOwnProperty("type")) var tokenType = token.type;
        if(token.hasOwnProperty("name")) var name= token.name;
        if(token.hasOwnProperty("value")) var value = token.value;
        
        const restrictedNames = ['debug','debug_propagate_exceptions'];


        if(tokenType == "variable" && restrictedNames.includes(name.toLowerCase()) && value == true){

            const warning = 'possible debug set true at line '+ lineno;
            vscode.window.showWarningMessage(warning);
            // vscode.commands.executeCommand('revealLine',{'lineNumber':lineno, 'at':'top'});
        }
        else if(tokenType == "function_call") {
            if(token.hasOwnProperty("keywords")) var keywords = token.keywords;
            
            keywords.map(keyword => {
                if(restrictedNames.includes(keyword[0].toLowerCase()) && keyword[1] == true){

                    const warning = 'possible debug set true at line '+ lineno;
                    vscode.window.showWarningMessage(warning);
                    // vscode.commands.executeCommand('revealLine',{'lineNumber':lineno, 'at':'top'});
                }
            });
        }
        else if(tokenType == 'dict' && token.hasOwnProperty('keys') && token.hasOwnProperty('values')){
            var keys = token.keys;
            var values = token.values;
            
            if(keys.length == values.length)
            {
                var blocks = keys.map((key, i) => [key, values[i]]);
    
                blocks.map(block => {
                    if(restrictedNames.includes(block[0].toLowerCase()) && block[1] == true) {
                        const warning = 'possible debug set true at line '+ lineno;
                        vscode.window.showWarningMessage(warning);
                    }
                });
            }
        }
    }
}

module.exports = smell;

File: py/smells/debugflag.py
from operations.savewarnings import saveWarnings

def detect(token):

    if token.__contains__("line"): lineno = token["line"]
    if token.__contains__("type"): tokenType = token["type"]
    if token.__contains__("name"): name = token["name"]
    if token.__contains__("value"): value = token["value"]
    
    restrictedNames = ['debug', 'DEBUG', 'DEBUG_PROPAGATE_EXCEPTIONS']
    
    if tokenType == "variable" and name in restrictedNames and value:
        warning = 'debug set true'
        saveWarnings(warning,str(lineno))
        print(warning+ ' at line '+ str(lineno))

    elif tokenType == "function_call":
        if token.__contains__("keywords"): keywords = token["keywords"]
        
        for keyword in keywords:
            if(keyword[0] in restrictedNames and keyword[1]):
                warning = 'debug set true'
                saveWarnings(warning,str(lineno))
                print(warning+ ' at line '+ str(lineno))
